### ISSUE TITLE
generate accession identifiers plugin will break in future releases due to removal of Sequence class

### ISSUE THREAD

**Issue by lorawoodford** [archivesspace/archivesspace#3255]
The removal of the `Sequence` model in #3203 breaks the [generate_accession_identifiers plugin](https://github.com/archivesspace/archivesspace/tree/master/plugins/generate_accession_identifiers).  I don't know if this plugin is actually supported anymore and/or if any ArchivesSpace deployments currently make use of this plugin, but passing along this accidental discovery.  (For the record, we don't use/don't intend to use this plugin, so this is just informational.)

While this plugin is in the core codebase, it it not bundled into built releases of ArchivesSpace, so this falls into somewhat of a gray area.  If it is intended to be a deprecated plugin example, it should probably be moved out of the codebase and the `sequence` table dropped from the db, as folks - like me! - go looking there for exemplars of good plugins to borrow from/modify for specific needs.  If that is the will/decision of the program team, I'm happy to work on a PR to do that, but that is a decision that needs to be made at a program team level.

TLDR; whether intentional or not, 3.5.1 and onward will deprecate this plugin and create an orphaned table in the db.

## Expected Behavior

Enabling the `generate_accession_identifiers` in `config/config.rb` plugin should just work and accession numbers should autogenerate and increment as designed.

## Current Behavior

Enabling the plugin does not autogenerate and increment accession numbers (you must manually enter accession numbers).  If you attempt to post to the [plugin's endpoint](https://github.com/archivesspace/archivesspace/blob/master/plugins/generate_accession_identifiers/backend/controllers/generate_4part_id.rb) you get `uninitialized constant ArchivesSpaceService::Sequence`.

## Possible Solution
Remove the plugin and alert the community/update 3.5.1 release notes to state that plugin will no longer be supported

OR 

Fix the plugin (including restoring the `Sequence` class) -- though I expect there were good Java-upgrade related reasons why this was removed in the first place, so I don't know what the level of effort involved would be there?

## Steps to Reproduce (for bugs)
1. Enable plugin, create an accession -- confirm accession number is not auto-generated
2. Hit the `/plugins/generate_accession_identifiers/next` API endpoint and confirm error

## Context
We are unaffected, but figured this should be raised/documented somewhere

## Your Environment
* Version used: current master branch (plugin confirmed to be working in all releases 3.5.0 and prior, but not in release 3.5.1 or any commits after #3203
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):
* Link to your project:

**Comment by jdshaw** [comment]
We use a forked version of the plugin and I know of a couple of other institutions who also use customized versions. While I *can* add the model back in in our plugin, having it in core would be handy.
